File: python/functional.py
def runFunctionalDemo():
    nums = [1, 2, 3, 4, 5]
    mapped = list(map(lambda x: x * 2, nums))
    filtered = list(filter(lambda x: x % 2 == 0, nums))
    from functools import reduce

    reduced = reduce(lambda acc, x: acc + x, nums)

    print("Map:", mapped)
    print("Filter:", filtered)
    print("Reduce:", reduced)

File: csharp/Functional.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Functional
{
    public static void runFunctionalDemo()
    {
        Console.WriteLine("\n🧪 Functional Demo");
        var nums = new List<int> { 1, 2, 3, 4, 5 };
        var mapped = nums.Select(x => x * 2).ToList();
        var filtered = nums.Where(x => x % 2 == 0).ToList();
        var reduced = nums.Aggregate((acc, x) => acc + x);

        Console.WriteLine("Map: " + string.Join(", ", mapped));
        Console.WriteLine("Filter: " + string.Join(", ", filtered));
        Console.WriteLine("Reduce: " + reduced);
    }
}
